### DISCUSSION TITLE
Best practice around running a batch job to insert new rows daily

### DISCUSSION THREAD

**Discussion by manas007** [milvus-io/milvus#31261] [Q&A and General discussion]
Hello,
I have recently started playing around with milvus using the python SDK and it has been going well so far. I can read data from a snowflake table, vectorize it, and insert it into a collection. 
Question: what are the best practices on setting up a pipeline to do this job every day? 
Pretty sure its a fairly simple use case and folks have done it in the past. What have you used to create the ETL pipeline to do this?
Thanks.

**Comment by yhmo** [comment]
No preference here. You can use any tool/framework that you are familiar with to do the work.

Optional, there is a tool for "ETL for unstructured data" built by Milvus team: https://docs.towhee.io/

**Comment by xiaofan-luan** [comment]
> Hello, I have recently started playing around with milvus using the python SDK and it has been going well so far. I can read data from a snowflake table, vectorize it, and insert it into a collection. Question: what are the best practices on setting up a pipeline to do this job every day? Pretty sure its a fairly simple use case and folks have done it in the past. What have you used to create the ETL pipeline to do this? Thanks.

One of the recommendation is to use Spark connector. We have a example to export data from mysql and bulkinsert with mysql.

**Comment by xiaofan-luan** [comment]
see those use cases at https://milvus.io/docs/integrate_with_spark.md.

If you are a Zilliz cloud user, we actually working a bunch of connectors that convert different sources to Zilliz cloud.
we can talk about this further if you are interested, ping me @ xiaofan.luan@zilliz.com

**Reply by manas007** [reply]
Thanks. Not a cloud user. I am just toying around, building a web app. High level use case is source data from snowflake, vectorize it and store in milvus. send a query to milvus from the web app when user requests to see items similar to a selected item. hope that makes sense.
